### PR TITLE
MAINT: version bounds for 1.10.0rc1/relnotes fixes

### PR DESCRIPTION
**Pull request by tylerjereddy**
* the version bound changes are very easy to mess up; the conversation from the last release is available here:
https://github.com/scipy/scipy/pull/16353

* at the moment, I've assumed that NumPy `1.24.0` will be released before SciPy `1.10.0`, so that we can perhaps aim to support all the way up to `1.26.0` if we are careful?

* one other point of caution--we rely even more on `meson` now than we did at the start of the `1.9.0` release cycle for things like wheel builds, so if you think of anything else that might need adjusting in that regard let me know

* I think I've traditionally left the adjustment of some version strings to include `rc1` to the release process proper/`REL` commit

* clean up some release notes issues: remove some empty sections, fix the formatting of various nested lists; this is made a bit easier by pasting sections into online RST interactive preview pages--let me know if you see other parts that need fixing

* the full testsuite passed locally, but I'll flush the full CI for now because it it useful to know that the maintenance branch CI doesn't have any surprises waiting

### CODE REVIEW

**Comment by mdhaber** [comment]
Thanks, Tyler. I submitted tylerjereddy/scipy#63 with some more stats fixes and improvements.

File: meson.build [hunk 1]
```diff
@@ -6,7 +6,7 @@ project(
   # tools/version_utils.py
   version: '1.10.0.dev0',
   license: 'BSD-3',
-  meson_version: '>= 0.64.0',
+  meson_version: '>= 0.64.0,< 0.66',
```
Comment (rgommers) [inline meson.build:9]: Unfortunately you cannot do this. I ran into that once before. It causes this in CI:
```
meson.build:1:0: ERROR: Meson version is 0.64.0 but project requires >= 0.64.0,< 0.66
```

**Review by rgommers — COMMENTED** [review]
LGTM overall, a few small comments

**Comment by tylerjereddy** [comment]
I merged Matt's `stats` section release notes adjustments and pushed in a commit to address Ralf's comments so far.

If the CI turns green, the docs render looks "ok" for release notes, and I don't see any other change requests, I'll probably proceed with the RC1 release process starting this evening or this weekend.

**Comment by tylerjereddy** [comment]
Release note still has the `X` in:

> SciPy 1.10.0 is the culmination of X months of hard work. 

I'll push that fix in, scoping CI to only rebuild docs, then I'll move forward.

**Comment by tylerjereddy** [comment]
`1.10.0rc1` wheels are now building at the links below. @andyfaff I noticed a connection issue/failure on 1/4 of the cirrus CI tasks (https://cirrus-ci.com/task/6403987432275968) but restarting manually in the Cirrus CI interface seems to have done the trick.

If I see any areas where we can improve things on the wheel building I'll open an issue, but so far so good.

Ongoing wheel builds: 

- GHA: https://github.com/scipy/scipy/actions/runs/3665807972/jobs/6197132229
- Cirrus: https://cirrus-ci.com/build/5784600870387712

**Comment by mdhaber** [comment]
Thanks, Tyler! I noticed one thing I missed in my PR - perhaps we can add it for the final release:
image

That should have been `scipy.stats.mstats.hdquantiles_sd`.

**Comment by tylerjereddy** [comment]
ok, I've made a note to fix that on rc2

**Pull request merged**